llvm/lib/Target/NVPTX/NVPTXLowerAlloca.cpp

```diff
@@ -72,12 +72,26 @@ bool NVPTXLowerAlloca::runOnFunction(Function &F) {
         Changed = true;
         auto ETy = allocaInst->getAllocatedType();
         auto LocalAddrTy = PointerType::get(ETy, ADDRESS_SPACE_LOCAL);
-        auto NewASCToLocal = new AddrSpaceCastInst(allocaInst, LocalAddrTy, "");
-        auto GenericAddrTy = PointerType::get(ETy, ADDRESS_SPACE_GENERIC);
-        auto NewASCToGeneric =
-            new AddrSpaceCastInst(NewASCToLocal, GenericAddrTy, "");
-        NewASCToLocal->insertAfter(allocaInst);
-        NewASCToGeneric->insertAfter(NewASCToLocal);
+        PointerType *AllocInstPtrTy =
+            cast<PointerType>(allocaInst->getType()->getScalarType());
+        Instruction *NewASCToGeneric = allocaInst;
```

Comment:
This could use a better name, given that ASC may or may not be there now. `AllocaInGenericAS` ? 
I'd also push the variables down, into the `if` body -- some of them seem to be needed by the now-conditional code only.

Reply:
Hmmm, this is original name in the source code, happy to rename it if that's better.

Reply:
You're changing what the code does, Adjusting variable names as appropriate is the right thing to do.